src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -458,6 +464,15 @@ public void StopStake()
 
             while (!this.stakeCancellationTokenSource.Token.IsCancellationRequested)
             {
+                // Prevent mining if the system time is not in sync with that of other members on the network.
+                if (this.timeSyncBehaviorState.IsSystemTimeOutOfSync())
+                {
+                    this.logger.LogError("Staking cannot start, your system time does not match that of other nodes on the network." + Environment.NewLine
+                                         + "Please adjust your system time and restart the node.");
+                    await Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), this.stakeCancellationTokenSource.Token).ConfigureAwait(false);
+                    continue;
```

Comment:
I don't think this is correct. What this does is that every half a second it displays an error on a console. That seems like overkill and it will be mixed with those other messages that you display from other components so it will be quite messy. Instead of this continue maybe you would like to Implement some kind of mechanism that would actually terminate stakingLoop , Or maybe at least just wait a lot longer than half a second

Reply:
That's the intended purpose: making it really clear to the user that his node isn't staking. 
If we just stop staking and show a message that says as much, this message will quickly disappear amidst the other logs.
We could extend it to be every 5 seconds I guess.

Reply:
There are other things happening every 5 seconds so you probably want to have something different maybe 7 seconds, maybe 11 seconds, not 5, and no less than that

Reply:
gotcha